internal/handler/ownership_proof_handler.go

```diff
@@ -0,0 +1,360 @@
+	}
+
+	err = o.ownershipProofService.Delete(dormID)
+	if err != nil {
+		return err
+	}
+
+	return c.SendStatus(fiber.StatusNoContent)
+
+}
+
+// Update godoc
+// @Summary Update ownership proof
+// @Description Replace an existing ownership proof file
+// @Tags ownership
+// @Security Bearer
+// @Accept multipart/form-data
+// @Produce json
+// @Param file formData file true "New ownership proof file"
+// @Param dormId formData string true "Dorm ID (UUID format)"
+// @Success 200 {object}  dto.SuccessResponse[dto.OwnershipProofWithFileResponseBody] "Ownership proof updated"
+// @Failure 400 {object} dto.ErrorResponse
+// @Failure 500 {object} dto.ErrorResponse
+// @Router /ownership/update [post]
+func (o *OwnershipProofHandler) Update(c *fiber.Ctx) error {
+
+	//extract file from http
+	file, err := c.FormFile("file")
+	if err != nil {
+		return apperror.BadRequestError(err, "file is required")
+	}
+
+	dormIDStr := c.FormValue("dormId")
+	dormID, err := uuid.Parse(dormIDStr)
+
+	if err != nil {
+		return apperror.BadRequestError(err, "Dorm id is required")
+	}
+
+	fileData, err := file.Open()
+	if err != nil {
+		return apperror.InternalServerError(err, "error opening file")
+	}
+	defer fileData.Close()
+
+	ownershipProof, err := o.ownershipProofService.GetByDormID(dormID)
+	if err != nil {
+		return err
+	}
+	fileKey := ownershipProof.FileKey
+	deleteErr := o.storage.DeleteFile(c.Context(), fileKey, storage.PrivateBucket)
+
+	if deleteErr != nil {
+		if apperror.IsAppError(deleteErr) {
+			return deleteErr
+		}
+		return apperror.InternalServerError(deleteErr, "error deleting file")
+	}
+	//prepare element for upload to storage
+	filename := strings.ReplaceAll(file.Filename, " ", "-")
+	contentType := file.Header.Get("Content-Type")
+	file_uuid := uuid.New().String()
+	newFileKey := fmt.Sprintf("ownership-proof/%s-%s", filename, file_uuid)
+
+	//upload to storage
+	err = o.storage.UploadFile(c.Context(), newFileKey, contentType, fileData, storage.PrivateBucket)
+	if err != nil {
+		return apperror.InternalServerError(err, "error uploading file")
+	}
+
+	//get key file
+	url, err := o.storage.GetSignedUrl(c.Context(), newFileKey, time.Minute*60)
+	if err != nil {
+		return apperror.InternalServerError(err, "error getting signed url")
+	}
+
+	updateErr := o.ownershipProofService.UpdateDocument(dormID, newFileKey)
+	if updateErr != nil {
+		if apperror.IsAppError(updateErr) {
+			return updateErr
+		}
+		return apperror.InternalServerError(updateErr, "error updating file")
+	}
+
+	ownershipProof, getErr := o.ownershipProofService.GetByDormID(dormID)
+	if getErr != nil {
+		if apperror.IsAppError(getErr) {
+			return getErr
+		}
+		return apperror.InternalServerError(getErr, "error getting new file")
+	}
+	ownershipProofResponse := o.ownershipProofService.ConvertToDTOWithFile(*ownershipProof, url, time.Now().Add(time.Minute*60))
+
+	return c.Status(fiber.StatusOK).JSON(dto.Success(ownershipProofResponse))
+
+}
```

Comment:
The update is similar to the Create feature. please rethinking possibility to merge as one fuction.